Sources/SwiftDocCPluginUtilities/BuildGraph/DocumentationBuildGraph.swift

```diff
@@ -0,0 +1,108 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+import Foundation
+
+/// A target that can have a documentation task in the build graph
+protocol DocumentationBuildGraphTarget {
+    typealias ID = String
+    /// The unique identifier of this target
+    var id: ID { get }
+    /// The unique identifiers of this target's direct dependencies (non-transitive).
+    var dependencyIDs: [ID] { get }
+}
+
+/// A build graph of documentation tasks.
+struct DocumentationBuildGraph<Target: DocumentationBuildGraphTarget> {
+    fileprivate typealias ID = Target.ID
+    /// All the documentation tasks
+    let tasks: [Task]
+    
+    /// Creates a new documentation build graph for a series of targets with dependencies.
+    init(targets: some Sequence<Target>) {
+        // Create tasks
+        let taskLookup: [ID: Task] = targets.reduce(into: [:]) { acc, target in
+            acc[target.id] = Task(target: target)
+        }
+        // Add dependency information to each task
+        for task in taskLookup.values {
+            task.dependencies = task.target.dependencyIDs.compactMap { taskLookup[$0] }
+        }
+        
+        tasks = Array(taskLookup.values)
+    }
+    
+    /// Creates a list of dependent operations to perform the given work for each task in the build graph.
+    ///
+    /// You can add these operations to an `OperationQueue` to perform them in reverse dependency order
```

Comment:
I have normally heard this ordering refered to as "dependency order" and "reverse dependency order" used to refer to the inverse, i.e. (dependents before their dependencies.)
```suggestion
    /// You can add these operations to an `OperationQueue` to perform them in dependency order
```

Reply:
Fixed in https://github.com/swiftlang/swift-docc-plugin/pull/84/commits/7ecba951ab8908b049c0eaf0fb4071af5bcf4789